### PR TITLE
Autonavigation

### PR DESCRIPTION
**Pull request by narayanan-iyer-pfizer**
This addresses #644  and  adds new buttons

### CODE REVIEW

File: R/mod_code_explorer.R [hunk 1]
```diff
@@ -167,6 +184,46 @@ mod_code_explorer_server <- function(id, selected_pkg, pkgarchive = reactiveVal(
     
     introJSServer("introJS", text = reactive(fe_steps), user, credentials)
     
+    observeEvent(input$next_button,{
+      if (input$next_button > 0){
+      shinyjs::runjs('
+                    var $index =Array.from($highlights_list).findIndex(node => node.isEqualNode($curr_sel));
```
Comment (jthompson-arcus) [inline R/mod_code_explorer.R:190]: This line appears to not be working.

**Review by jthompson-arcus — CHANGES_REQUESTED** [review]
@narayanan-iyer-pfizer thanks for working on this. The feature is not working for me. Didn't dig too deep into why that was the case. The JavaScript does not appear to be functioning as intended though.

I'm not a huge fan of how the buttons look. I think personally I would want something more minimal. Could a count also be included (e.g. < 1 of 3 >)? I think that would add a lot.

File: R/mod_code_explorer.R (hunk 1 above)
Comment (narayanan-iyer-pfizer) [reply R/mod_code_explorer.R:195]: fixed and count added

**Comment by jthompson-arcus** [comment]
@narayanan-iyer-pfizer I started looking at this again and then realized you didn't re-request me to review. Did you want a review or are you still working?

**Comment by narayanan-iyer-pfizer** [comment]
> @narayanan-iyer-pfizer I started looking at this again and then realized you didn't re-request me to review. Did you want a review or are you still working?

Please review it. I'm done with this

**Review by jthompson-arcus — CHANGES_REQUESTED** [review]
@narayanan-iyer-pfizer sorry for sitting on this review for so long. 

In addition to the conversations below, I noticed that if you select the file type "Help Documentation" that the buttons stay visible for the last view.
image width="100%"

File: R/mod_code_explorer.R [hunk 2]
```diff
@@ -31,7 +31,6 @@ mod_code_explorer_server <- function(id, selected_pkg, pkgarchive = reactiveVal(
         showHelperMessage(message = glue::glue("Source code not available for {{{selected_pkg$name()}}}"))
       } else {
         div(introJSUI(NS(id, "introJS")),
-          br(),
           fluidRow(
```
[the remark this answers is not in the document]
Comment (narayanan-iyer-pfizer) [reply R/mod_code_explorer.R:35]: done

**Pull request merged**